posthog/api/test/test_trends.py

```diff
@@ -112,30 +116,106 @@ def test_includes_only_intervals_within_range(client: Client):
         }
 
 
+@pytest.mark.django_db
+@pytest.mark.ee
+def test_math_is_functional(client: Client):
```

Comment:
What about testing this on postgres?

Also now we have 3 different ways to write tests for queries, each of which has its own strengths which is gonna lead to more issues. What's the plan here?

Reply:
Yup I'll check on postgres.

Yes that is troubling.

For this specific case, where we are testing a specific user scenario I found it useful to go through the process of:

 1. reproducing the issue on app.posthog.com
 2. verifying that it's not a UI issue by poking around in the code
 3. identifying the query that is being sent to the api
 4. writing this test using the API request with my expectation

This seemed like a very simple flow, but I can see that it would also make sense to add tests for the specific module that might be shared by other parts of the system.

Reply:
ways of writing this type of test:

1. with `django.test.TestCase`
   2. with self.client
   3. using `...Query` objects
2. with pytest function test using DjangoClient

Note: you can use `pytest --log-cli-level=DEBUG`